src/openrct2/actions/PlaceParkEntranceAction.hpp

```diff
@@ -93,10 +93,11 @@ DEFINE_GAME_ACTION(PlaceParkEntranceAction, GAME_COMMAND_PLACE_PARK_ENTRANCE, Ga
                 entranceLoc.y += CoordsDirectionDelta[(_loc.direction + 1) & 0x3].y * 2;
             }
 
-            if (!map_can_construct_at({ entranceLoc, zLow, zHigh }, { 0b1111, 0 }))
+            if (auto res2 = MapCanConstructAt({ entranceLoc, zLow, zHigh }, { 0b1111, 0 }); res2->Error != GA_ERROR::OK)
```

Comment:
I'd prefer if the assignment would take place outside the if. I think this is hard to read (and also not consistent with the rest of the code).

Reply:
the result is only useful for the inside of the if statement. It doesn't seem that useful bringing it out into the larger scope of the function.